### PR TITLE
iOS: Change release artifacts to use xcframeworks

### LINKED ISSUE #2215: CocoaPods 0.4.6 release does not compile
There's something wrong with the podspec configuration.

### PR DESCRIPTION
**Pull request by jpsim**
Fixes #2215 & #2216

See [discussion on Slack](https://envoyproxy.slack.com/archives/CKQ2LK23G/p1651002845920269).

Risk Level: Low to existing users since the previous frameworks don't appear to have really been working, or been in use
Testing: [Validated on CI](https://github.com/envoyproxy/envoy-mobile/actions/runs/2229074086), locally with `pod spec lint` pointing to the new zip format, locally with SwiftPM, locally by dragging & dropping the xcframework into an Xcode project.
Docs Changes: None
Release Notes: Added

Signed-off-by: JP Simard <jp@jpsim.com>

### CODE REVIEW

File: library/swift/resources/BUILD [hunk 1]
```diff
@@ -0,0 +1,6 @@
+licenses(["notice"])
+
+exports_files(
```
Comment (keith) [inline library/swift/resources/BUILD:3]: if you just don't have a BUILD file in this dir you can reference this file directly. you just can't do that across package boundaries

**Comment by jpsim** [comment]
I measured the final app size impact of integrating Envoy Mobile in an iOS app to be ~5MB when statically linked and ~9MB when dynamically linked.

We can optimize binary size further in the future, but for now I believe that moving to dynamic xcframeworks gives the best developer experience by resolving the issues described in https://github.com/envoyproxy/envoy-mobile/issues/2215 & https://github.com/envoyproxy/envoy-mobile/issues/2216.

**Comment by jpsim** [comment]
I changed to use static xcframeworks in the last commits, since I got that working after pulling in https://github.com/bazelbuild/rules_apple/pull/1443 and updating the bazel version to `6.0.0-pre.20220414.2`.

I started a conversation in the Envoy Slack about adopting rolling bazel releases in general: https://envoyproxy.slack.com/archives/C7E6C71QB/p1651089150478729

It would be nice to stay in sync with Envoy to avoid any incompatibilities, but we used to have separate bazel versions prior to https://github.com/envoyproxy/envoy-mobile/pull/1925 and I think if we get clear wins from updating sooner it should be considered.

**Comment by jpsim** [comment]
Seems like our java runtime setup broke somewhere in https://github.com/bazelbuild/bazel/compare/6.0.0-pre.20220208.2...6.0.0-pre.20220216.3

**Comment by keith** [comment]
Yea I hit that locally when testing this as well, there are some flags in the .bazelrc, I think ideally we would be passing `--java_runtime_version=remotejdk_11 --tool_java_runtime_version=remotejdk_11` but I haven't debugged more

**Comment by jpsim** [comment]
Another benefit of moving to xcframeworks is that we get Swift Package Manager support for free.

```swift
.binaryTarget(
  name: "Envoy",
  url: "https://github.com/envoyproxy/envoy-mobile/releases/download/v0.4.6/Envoy.xcframework.zip",
  checksum: "..."
)
```

**Comment by jpsim** [comment]
> Seems like our java runtime setup broke somewhere in [bazelbuild/bazel@6.0.0-pre.20220208.2...6.0.0-pre.20220216.3](https://github.com/bazelbuild/bazel/compare/6.0.0-pre.20220208.2...6.0.0-pre.20220216.3)

Bisected the failure to https://github.com/bazelbuild/bazel/commit/d5559c16ac008b86345fbbade5d600181a2fce6f

I don't have any ideas to explore to resolve the issue. Passing `--java_runtime_version=remotejdk_11 --tool_java_runtime_version=remotejdk_11` doesn't appear to help.

**Comment by keith** [comment]
Thanks that was super helpful, I believe this fixes that issue: https://github.com/envoyproxy/envoy-mobile/pull/2227

**Comment by jpsim** [comment]
With `main` now on bazel 6.x, this is ready for review without any egregious hacks.

**Review by keith — APPROVED** [review]

**Review by Augustyniak — DISMISSED** [review]

**Comment by jpsim** [comment]
I'm waiting on #2184 to merge this to not cause more work in that PR.

**Review by keith — APPROVED** [review]

**Comment by jpsim** [comment]
I didn't merge the `release.yml` GitHub Actions workflow properly. Updating.

**Review by keith — APPROVED** [review]

**Pull request merged**